lib/autocomplete-provider.js

```diff
@@ -1,74 +1,257 @@
+    if (!/( ::|data |class |type |newtype )/.test(content)) {
+      return [];
+    }
+
+    // Parse and clean output from ghc.browseTidal()
+    let functions;
+    let parts = content.split(/\n> ([\s\S]*)/);
+
+    if (parts.length < 2) {
+      functions = content.split('\n');
+    } else {
+      functions = parts[1].split('\n');
+      functions = functions.slice(0, functions.length - 2);
+    }
+
+    functions = functions
+      .reduce((acc, cur) => {
+        if (cur.startsWith(" ")) {
+          return acc + ' ' + cur.trim();
+        } else {
+          return acc + '\n' + cur.trim();
+        }
+      }, '')
       .split("\n")
-      .reduce((acc, cur) =>
-        cur.startsWith(" ")
-          ? acc + ' ' + cur.trim()
-          : acc + '\n' + cur
-      )
+      .filter(line => line.length > 0)
+      .filter(line => !line.startsWith(':'))
+      .filter(line => !/^[A-Za-z]*>/.test(line));
 
+    let seen = new Set();
     return functions
-      .split("\n")
-      .filter(row => row.indexOf('::') > -1)
       .map(row => {
-        let fields = row.split('::')
-        let functionPath = fields[0].trim().replace("(", "").replace(")", "")
-        let functionName = functionPath.substring(functionPath.lastIndexOf('.') + 1)
-        let docsURL = `https://tidalcycles.org/search?q=${encodeURIComponent(functionName)}`
-        let hoogleURL = `https://hoogle.haskell.org/?hoogle=${encodeURIComponent(functionName)}&scope=package:tidal`
+        row = row.trim();
+        let formatted = this.formatTypeSignature(row);
+
+        if (!formatted) {
+          return null;
+        }
+
+        let [functionPath, typeSignature] = formatted;
+        let functionName = functionPath.substring(functionPath.lastIndexOf('.') + 1);
+
+        // Filter out kind signatures (containing only *, :, ->, =>). There are other definitions for these types.
+        if (/^[\s*:>\-=]+$/.test(typeSignature)) {
+          return null;
+        }
+
+        let docsURL = `https://tidalcycles.org/search?q=${encodeURIComponent(/[A-Za-z0-9]/.test(functionName) ? functionName : (functionName + ' operator'))}`;
+        let hoogleURL = `https://hoogle.haskell.org/?hoogle=${encodeURIComponent(functionName)}&scope=package:tidal`;
+
         return {
-          text: functionPath.substring(functionPath.lastIndexOf('.') + 1),
+          text: functionName,
           snippet: `${functionName} `,
-          description: fields[1].trim(),
+          description: typeSignature,
           type: 'function',
           rightLabel: functionPath.substring(0, functionPath.lastIndexOf('.')),
           descriptionMoreURL: hoogleURL,
           leftLabelHTML: `<a data-docs-link href="${docsURL}" title="Open tidalcycles.org in a browser." tabindex="-1">🔗</a>`
-        }
+        };
       })
+      .filter(item => {
+        if (item === null) return false;
+        if (seen.has(item.text)) return false;
+        seen.add(item.text);
```

Comment:
maybe this `seen` evaluation could be moved in the very moment we create the `functionName`, that's the variable we want to deduplicate, so, after line 230 there could be this code:
```js
...
let functionName = functionPath.substring(functionPath.lastIndexOf('.') + 1);

if (seen.has(functionName)) return null;
seen.add(functionName);
```

and this filter just filters out null items

Reply:
Done, that's a better choice.